src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1983,10 +2077,32 @@ instance Semigroup LimitClause where
 
 instance Monoid LimitClause where
   mempty = Limit mzero mzero
-  mappend = (<>)
 
 -- | A locking clause.
-type LockingClause = Monoid.Last LockingKind
+data LockingClause =
+  GeneralLockingClause LockingKind
+  -- ^ Locking clause not specific to any database implementation
+  | PostgresLockingClauses [PostgresLockingKind]
+  -- ^ Locking clause specific to postgres
+  | NoLockingClause
+
+instance Semigroup LockingClause where
+  -- Postgres allows us to have multiple locking clauses
+  (<>) (PostgresLockingClauses pleft) (PostgresLockingClauses pright) = PostgresLockingClauses (pleft <> pright)
+  -- For backwards compatibility reasons we use the monoid instance of last for
+  -- general locking cluase
+  (<>) (GeneralLockingClause gleft) (GeneralLockingClause gright)  =
+    Maybe.maybe mempty GeneralLockingClause
+      (Monoid.getLast $ Monoid.Last (Just gleft) <> Monoid.Last (Just gright))
```

Comment:
I think that `getLast (Last _ <> Last b)` will always be `Last b`, so we can probably simplify this 

```suggestion
    GeneralLockingClause gright
```